elasticsearch/helpers/vectorstore/_async/vectorstore.py

```diff
@@ -306,6 +308,10 @@ async def _create_index_if_not_exists(self) -> None:
             vector_field=self.vector_field,
             num_dimensions=self.num_dimensions,
         )
+        if self.custom_settings:
+            self.custom_settings.update(settings)
```

Comment:
The way you coded this means that if the same setting is provided by the user and set by the strategy, then the value from the strategy wins. Is this what you intended? To me it seems more reasonable to do the reverse, which would allow the caller to override settings.

Either way, the behavior should be documented in the docstring above.

Reply:
I believe the settings in the strategy should have a higher priority because I am concerned that users might overlook the default strategy settings, which could lead to unknown errors. Of course, I also respect your possible different opinion, and if you think it's necessary to make other changes, please let me know.

Reply:
@pquentin do you have an opinion on this?

I'm not sure I like the idea that a custom setting that you pass is silently ignored if the selected strategy uses a different value, first because it is silent so the developer may not realize a setting isn't being applied, and second because it provides to option to override the settings in the strategy.

The ideal solution would be:

- if there is a collision between settings passed by the user and by the strategy, then raise an error
- add another boolean argument that the user can set to `True` to indicate that overriding of strategy settings should be allowed, in which case the error will not be raised and the custom settings from the user take precedence.

Reply:
Yes, I also think that if you specify a setting, it should be used and not silently discarded as done now.

I'm happy to raise an error if there's a collision, but I don't think we should bother with another boolean argument. The strategy settings exist for a reason, and the custom user settings are going to be quite different anyway (number of primary/replicas, etc.). For this reason, we can probably limit collision detection to top-level settings keys for now. And all of this can be reevaluated if we have use case that requires more complexity.

Reply:
@minglu7 Let's go with @pquentin simplified proposal for now.

- If a top-level setting is in both groups of settings, then raise an error
- Else, a merge can be done cleanly so no problem.

In the future, we can re-evaluate if an override option is needed, but for now let's avoid this complication.

Reply:
@miguelgrinberg i have did these changes as you wish, please check them